compiler/qsc/src/bin/qsi.rs

```diff
@@ -72,12 +73,20 @@ impl Receiver for TerminalReceiver {
 
 fn main() -> miette::Result<ExitCode> {
     let cli = Cli::parse();
-    let sources = cli
+    let mut sources = cli
         .sources
         .iter()
         .map(read_source)
         .collect::<miette::Result<Vec<_>>>()?;
 
+    let fs = StdFs;
+    let manifest = Manifest::load()?;
+    if let Some(manifest) = manifest {
+        let project = fs.load_project(manifest)?;
+        let mut project_sources = project.sources;
+
+        sources.append(&mut project_sources);
```

Comment:
Same here, looks like sources from the project manifest are appended regardless of whether other sources were provided as input.